pkg/resources/stream_on_table.go

```diff
@@ -0,0 +1,307 @@
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW STREAMS` for the given stream.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowStreamSchema,
+		},
+	},
+	DescribeOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `DESCRIBE STREAM` for the given stream.",
+		Elem: &schema.Resource{
+			Schema: schemas.DescribeStreamSchema,
+		},
+	},
+	FullyQualifiedNameAttributeName: schemas.FullyQualifiedNameSchema,
+}
+
+func StreamOnTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateStreamOnTable(false),
+		ReadContext:   ReadStreamOnTable(true),
+		UpdateContext: UpdateStreamOnTable,
+		DeleteContext: DeleteStreamContext,
+		Description:   "Resource used to manage streams on tables. For more information, check [stream documentation](https://docs.snowflake.com/en/sql-reference/sql/create-stream).",
+
+		CustomizeDiff: customdiff.All(
+			ComputedIfAnyAttributeChanged(streamOnTableSchema, ShowOutputAttributeName, "table", "append_only", "comment"),
+			ComputedIfAnyAttributeChanged(streamOnTableSchema, DescribeOutputAttributeName, "table", "append_only", "comment"),
+		),
+
+		Schema: streamOnTableSchema,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportStreamOnTable,
+		},
+	}
+}
+
+func ImportStreamOnTable(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] Starting stream import")
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	v, err := client.Streams.ShowByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if err := d.Set("name", id.Name()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("database", id.DatabaseName()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("schema", id.SchemaName()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("append_only", booleanStringFromBool(v.IsAppendOnly())); err != nil {
+		return nil, err
+	}
+	return []*schema.ResourceData{d}, nil
+}
+
+func CreateStreamOnTable(orReplace bool) schema.CreateContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		databaseName := d.Get("database").(string)
+		schemaName := d.Get("schema").(string)
+		name := d.Get("name").(string)
+		id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
+
+		tableIdRaw := d.Get("table").(string)
+		tableId, err := sdk.ParseSchemaObjectIdentifier(tableIdRaw)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		req := sdk.NewCreateOnTableStreamRequest(id, tableId)
+		if orReplace {
+			req.WithOrReplace(true)
+		}
+
+		if v := d.Get("copy_grants"); v.(bool) {
+			req.WithCopyGrants(true).WithOrReplace(true)
+		}
+
+		if v := d.Get("append_only").(string); v != BooleanDefault {
```

Comment:
Please check new `booleanStringAttributeCreate` helper func and others (check user resource); they allow writing shorter create/update/read.

Reply:
Done.